compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUTileAndDistribute.cpp

```diff
@@ -249,10 +254,14 @@ struct LLVMGPUTileAndDistributePass
  private:
   // Distribute the workloads to warp if true otherwise distribute to threads.
   bool distributeToWarp = false;
+  GPUPromoteSharedMemPattern promoteSharedMemPattern =
```

Comment:
@ThomasRaoux there is prolly a cleaner way to handle this. I need to modify the promotion pattern between matmul and transpose. There are several options used to configure it in LLVMGPUTileAndDistributePass

Reply:
Yes I was thinking about making it a lambda passed to the function, however this doesn't play very well with the lit test mechanism. I think it is fine to go with this for now.

Reply:
Ok, I'll leave it as is.